composer.json

```diff
@@ -15,22 +15,22 @@
   },
   "require": {
     "drupal/guzzle_cache": "^2.0",
+    "kevinrob/guzzle-cache-middleware": "~3.3.1",
```

Comment:
Interesting, isn't this required due to guzzle_cache above? Also, FYI there's an API-compatible 3.0 release of guzzle_cache, so you could `||` that in the require line.

Reply:
> Interesting, isn't this required due to guzzle_cache above?

It is, although when I was working on this, it would pick up a version of `kevinrob/guzzle-cache-middleware` that implicitlyl depended on a recent version of `guzzlehttp/psr7` without expressing that properly in it's requirements, which led to situations, like here, where it was using recent API's in `guzzlehttp/psr7` that weren't actually present. More details here: https://github.com/Kevinrob/guzzle-cache-middleware/pull/143#issuecomment-877700383

The good news is that the author made a new release with the right requirements, so I'm going to try to remove this and see if we're all good.

Reply:
Nope. Had to stick with `~3.3.1`. What it comes down to is that `drupal/core-recommended` (which the test image is based on) pins `guzzlehttp/psr7` at 1.6.1, which 3.3.1 of `kevinrob/guzzle-cache-middleware` is the last version to be compatible with that.

Reply:
I think if you do `^3.3`, it will allow you to upgrade once the psr7 constraint is lifted. `~3.3.1` _adds_ a constraint preventing upgrading to 3.4 or higher.

I think!

Reply:
Yep, `~3.3.1` means it can't go >= `3.4`. That's intentional ATM as with something like `^3.3` composer wants to go to `3.4`, but `3.4` has a hidden dependency on a version of `guzzlehttp/psr7` that is too recent for `drupal/core-recommended` to allow. `3.4.1` has the hidden dependency declared explicitly, but that means composer won't take it b/c it conflicts with `drupal/core-recommended`. So we're stuck sticking to the `3.3.x` line until a major update unfortunately that shirks D8.

Reply:
> a hidden dependency on a version of `guzzlehttp/psr7`

_ohhhh_ ok, that makes sense. At a moment like this I really wish you could have comments in JSON, but at least we'll have this thread for future us to remember!

Reply:
Totally! I definately googled "comments in composer.json" along the way here :laughing: